### DISCUSSION TITLE
Passing authentication headers into transcribe_url method failing with DeepgramApiError: The remote server hosting the media returned a client error: 401 Unauthorized. (Status: 400)

### DISCUSSION THREAD

**Discussion by jasonthorpe** [deepgram/community#868] [General help]
I am trying to pull the wav files directly from a Vonage VCC system into Deepgram without having to download the file to my server and then subsequently upload it. Getting the files from the Vonage system with this method works fine:

```
    def get_call_audio(self, guid):
        token = AuthService.get_access_token(
            client_id=self.client_id, client_secret=self.client_secret, scope=self.scope
        )

        url = f"{self.base_url}/{guid}/content/callRecording"

        headers = {
            "Authorization": f"Bearer {token}",
            "Accept": "application/vnd.newvoicemedia.v3+json",
        }

        response = requests.get(url, headers=headers, stream=True)

        logging.debug(f"Response: {response}")

        if response.status_code == 200:
            return response
        else:
            response.raise_for_status()
```

However when I try to use the transcribe_url method instead of the transcribe_file passing in the Bearer token as below I am getting an DeepgramApiError: The remote server hosting the media returned a client error: 401 Unauthorized. (Status: 400)

```
def fetch_and_process_transcription_remote(self, call_record_instance):
    # ... existing code ...

    logger.debug(f"Audio URL: {audio_url}")
    logger.debug(f"PBX Headers: {pbx_headers}")

    pbx_headers = {
        "Authorization": f"Bearer {token}",
        "Accept": "application/vnd.newvoicemedia.v3+json",
    }

    source: UrlSource = {
        "url": audio_url,
        "headers": pbx_headers
    }

    logger.debug(f"Deepgram Source: {source}")

    try:
        response = self.dg_client.listen.prerecorded.v("1").transcribe_url(
            source=source,
            options=options
        )
        # ... rest of the method ...
```

I have also tried passing the headers in via:

```
     # Transcribe the audio file using the URL
      response = self.dg_client.listen.prerecorded.v("1").transcribe_url(audio_url, options, headers=headers)
```
 
How do I pass in the bearer token for Deepgram to pass on to Vonage?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.
